### PR TITLE
util: port a queue container from tiflow

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #40205

Problem Summary:

the queue container is merged into tiflow in https://github.com/pingcap/tiflow/pull/6297

I'll need a FIFO queue in future PR

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
/cc @hawkingrei @liuzix @crelax

Welcome back to open source community

**Comment by hawkingrei** [comment]
@lance6716 BTW, You implemented a FIFO queue. Will you implement more queues like priority queue or LIFO queue?

File: util/queue/chunkqueue.go [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+	defaultSizePerChunk = 1024
+	// minimumChunkLen is the minimum length of each chunk
+	minimumChunkLen = 16
+	// defaultPitchArrayLen is the default length of the chunk pointers array
+	defaultPitchArrayLen = 16
+)
+
+// ChunkQueue is a generic, efficient, iterable and GC-friendly queue.
+// Attention, it's not thread-safe.
+type ChunkQueue[T any] struct {
```
Comment (hawkingrei) [inline util/queue/chunkqueue.go:34]: ChunkQueue is based on the chunk. But I think it is called queue and it will be easier to understand.

**Comment by lance6716** [comment]
/hold

I'm considering use a channel as a FIFO queue 😂

**Pull request closed without merging**